test/d2l-applied-filters/d2l-applied-filters.html

```diff
@@ -101,6 +101,107 @@
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					assert.equal(clearFilters.hidden, true);
+				});
+				test('selecting a 4th filter reveals the clear filters button', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+					options[1].click();
+
+					afterNextRender(options[1], () => {
+						assert.equal(clearFilters.hidden, false);
+						done();
+					});
+				});
+				test('selecting a 4th filter then removing it hides the clear filters button', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+					options[1].click();
+
+					afterNextRender(options[1], () => {
+						options[1].deselect();
+
+						afterNextRender(options[1], () => {
+							assert.equal(clearFilters.hidden, true);
+							done();
+						});
+					});
+				});
+				test('clear filters button is visible when all filters are applied', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+
+					options[1].click();
+					options[2].click();
+					options[4].click();
+					options[5].click();
+					options[7].click();
+					options[8].click();
+
+					afterNextRender(options[1], () => {
+						assert.equal(clearFilters.hidden, false);
+						done();
+					});
+				});
+				test('clear filters button is hidden when 0 filters are applied', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+
+					options[0].deselect();
+					options[3].deselect();
+					options[6].deselect();
+
+					afterNextRender(options[0], () => {
+						assert.equal(clearFilters.hidden, true);
+						done();
+					});
+				});
+				test('clear filters button is hidden when 1 filters is applied', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+
+					options[0].deselect();
+					options[3].deselect();
+
+					afterNextRender(options[0], () => {
+						assert.equal(clearFilters.hidden, true);
+						done();
+					});
+				});
+				test('selecting a 4th filter & clicking clear filters fires a d2l-filter-dropdown-cleared event', function(done) {
+					const clearFilters = appliedFilters.shadowRoot.querySelector('#d2l-clear-filters-button');
+					const options = dropdown.querySelectorAll('d2l-filter-dropdown-option');
+
+					document.addEventListener('d2l-filter-dropdown-cleared', () => {
+						done();
+					});
```

Comment:
We probably want to have this event handler remove itself on being invoked, to ensure it doesn't outlive the test/invoke the `done` for this test in another context unintentionally, etc.

It could also be added to the rendered fixture, so it's cleaned up once the fixture is torn down. You'd want to assert that the event has `bubbles: true` and `composed: true` in that case (ie, consumers can handle it).

Another option could be to stub + spy `document.addEventListener`, and check that it was called for that event synchronously, then clean up the spy after the test completes.